### PR TITLE
Fixing encryption create with contentResponseOnWriteEnabled false

### PR DESCRIPTION
**Pull request by simplynaveen20**
Currently we are getting below exception on createItem as code is trying to decrypt the data for null response when contentResponseOnWriteEnabled  is false(default), this PR will fix it.

```
Caused by: java.lang.AssertionError
	at com.azure.cosmos.encryption.implementation.EncryptionProcessor.decrypt(EncryptionProcessor.java:353)
	at com.azure.cosmos.encryption.CosmosEncryptionAsyncContainer.lambda$createItemHelper$13(CosmosEncryptionAsyncContainer.java:383)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:125)
	at reactor.core.publisher.MonoPublishOn$PublishOnSubscriber.run(MonoPublishOn.java:181)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28)
```

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java [hunk 1]
```diff
@@ -339,6 +339,10 @@ public void encryptAndSerializeProperty(EncryptionSettings encryptionSettings, O
     }
 
     public Mono<byte[]> decrypt(byte[] input) {
+        if (input == null || input.length < 1) {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java:342]: nit:
`input.length < 1` seems a bit cryptic, also array len can never be negative, I would do `input.length == 0`

```suggestion
        if (input == null || input.length == 0) {
```

File: sdk/cosmos/azure-cosmos-encryption/src/test/java/com/azure/cosmos/encryption/EncryptionAsyncApiCrudTest.java [hunk 2]
```diff
@@ -123,6 +125,27 @@ public void createItemEncrypt_readItemDecrypt() {
         validateResponse(properties, responseItem);
     }
 
+    @Test(groups = {"encryption"}, timeOut = TIMEOUT)
+    public void createItemEncryptWithContentResponseOnWriteEnabledFalse() {
+        CosmosAsyncClient asyncClient = null ;
+        try {
+            asyncClient =
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-encryption/src/test/java/com/azure/cosmos/encryption/EncryptionAsyncApiCrudTest.java:132]: you don't need to create a new client to change `ContentResponseOnWriteEnabled`.

You can set `ContentResponseOnWriteEnabled` per operation. the benefit is you don't create a new client in the test .
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemRequestOptions.java#L261-L264


Generally we should try to re-use the clients when possible as client instantiation is time costly on the test. (will add more time to the test time). Not a big deal anyway.

**Review by moderakh — APPROVED** [review]
LGTM. some minor comments.
Thanks.

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java (hunk 1 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java]: done

File: sdk/cosmos/azure-cosmos-encryption/src/test/java/com/azure/cosmos/encryption/EncryptionAsyncApiCrudTest.java (hunk 2 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-encryption/src/test/java/com/azure/cosmos/encryption/EncryptionAsyncApiCrudTest.java]: done

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**